sklearn/ensemble/_base.py

```diff
@@ -167,6 +167,16 @@ def _make_estimator(self, append=True, random_state=None):
             elif getattr(estimator, "criterion", None) == "mae":
                 estimator.set_params(criterion="absolute_error")
 
+        # TODO: Remove in v1.2
+        # max_features = 'auto' would cause warnings in every call to
+        # Tree.fit(..)
+        if isinstance(estimator, BaseDecisionTree):
+            if getattr(estimator, "max_features", None) == "auto":
+                if is_classifier(estimator):
+                    estimator.set_params(max_features="sqrt")
+                else:
+                    estimator.set_params(max_features=1.0)
+
```

Comment:
To be extra safe since `BaseEnsemble` is used in many different estimators, we can be extra explicit and do this parameter change for the estimators with deprecated parameters:

```python
if getattr(estimator, "max_features", None) == "auto":
    if isinstance(estimator, DecisionTreeClassifier):
        estimator.set_params(max_features="sqrt")
    elif isinstance(estimator, DecisionTreeRegressor)):
        estimator.set_params(max_features=1.0)
```

Note that `ExtraTree*` is a child class of `DecisionTree*`, so the above works for the `ExtraTree*` estimators.